src/RCONServer.cpp

```diff
@@ -112,24 +149,28 @@ void cRCONServer::Initialize(cIniFile & a_IniFile)
 		return;
 	}
 	
-	// Read and initialize both IPv4 and IPv6 ports for RCON
-	bool HasAnyPorts = false;
-	AString Ports4 = a_IniFile.GetValueSet("RCON", "PortsIPv4", "25575");
-	if (m_ListenThread4.Initialize(Ports4))
-	{
-		HasAnyPorts = true;
-		m_ListenThread4.Start();
-	}
-	AString Ports6 = a_IniFile.GetValueSet("RCON", "PortsIPv6", "25575");
-	if (m_ListenThread6.Initialize(Ports6))
+	// Read the listening ports for RCON from config:
+	AStringVector Ports = ReadUpgradeIniPorts(a_IniFile, "RCON", "Ports", "PortsIPv4", "PortsIPv6", "25575");
+
+	// Start listening on each specified port:
+	for (auto port: Ports)
 	{
-		HasAnyPorts = true;
-		m_ListenThread6.Start();
+		UInt16 PortNum = static_cast<UInt16>(atol(port.c_str()));
```

Comment:
Also this seems like a good case to use `StringToInteger<UInt16>` which checks for bounds as well.